### ISSUE TITLE
KNeighborsTimeSeriesClassifier not alligned with documentation

### ISSUE THREAD

**Issue by Xhelas** [rtavenar/tslearn#86]
- The documentation specify that KNeighborsTimeSeriesClassifier works with variable lenght time series. It seems that it is not the case (tested with the documentation's example and other examples too.

- KNeighborsTimeSeriesClassifier does not have a one argument fit_predict method as shown in the documentation.

**Comment by rtavenar** [comment]
Hi @Xhelas ,

Thanks for raising this issue. I see two problems there:
- It does not make sense to have a single-argument `fit_predict` method for a supervised classification algo, so this should be changed in the docs, definitely (I guess it has to be changed here, I don't think it is referenced elsewhere)
- knn classifier should work with variable-length time series, so here the doc is correct and we have to work on the implem

So basically, the fix should be 2-steps:
- fix the docs: should be pretty easy, but I have to figure out why latest versions of the docs do not compile on read the docs
- fix the class: I'll have a look at it asap (hopefully this afternoon)

**Comment by rtavenar** [comment]
OK, so I remove the false claims from the docs, I paste it here to put it back when the problem is solved: 

```
Supervised classification
-------------------------

* :ref:`KNeighborsTimeSeriesClassifier <knn-clf>`

Example
~~~~~~~

.. code-block:: python

    from tslearn.neighbors import KNeighborsTimeSeriesClassifier
    clf = KNeighborsTimeSeriesClassifier(metric="dtw")
    predicted_labels = clf.fit(X, y).predict(X)
```

And we should definitely find a way to have kNN deal with variable length time series when (soft-)DTW is used as a metric, and similar thing for SVM+GAK.
We are working on it, but it may take some time to be fixed, sorry for that.
